tiledb/sm/query/test/unit_query_condition.cc

```diff
@@ -173,6 +174,825 @@ TEST_CASE("QueryCondition: Test char", "[QueryCondition][char_value]") {
   REQUIRE(query_condition.field_names().count(field_name) == 1);
 }
 
+TEST_CASE(
+    "QueryCondition: Test AST construction, basic",
+    "[QueryCondition][ast][api]") {
+  std::string field_name = "x";
+  int val = 5;
+  QueryCondition query_condition;
+  REQUIRE(
+      query_condition
+          .init(
+              std::string(field_name), &val, sizeof(int), QueryConditionOp::LT)
+          .ok());
+  CHECK(
+      tiledb::test::ast_node_to_str(query_condition.ast()) ==
+      "x LT 05 00 00 00");
```

Comment:
Here and in unit-query-ast -- could we do some tests with numbers that are large enough to fill all of the bytes of the value with non-zeros?  I think there is a general question about test coverage of the queries (the functionality, not just if the trees are being formed correctly).  For another story, but we really need to make sure all operators and data types get covered.

Reply:
I thought those tests were covered in unit-bytevector?

Reply:
https://app.shortcut.com/tiledb-inc/story/18278/create-more-comprehensive-testing-for-querycondition-ast-construction